sdk/identity/identity/src/credentials/environmentCredential.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+import { TokenCredential, isNode, RequestOptionsBase } from '@azure/core-http';
+import { IdentityClientOptions } from '../client/identityClient';
+import { ClientSecretCredential } from './clientSecretCredential';
+
+export class EnvironmentCredential implements TokenCredential {
+  private _credential?: TokenCredential = undefined;
+
+  constructor(options?: IdentityClientOptions) {
+    if (!isNode) {
+      throw 'EnvironmentCredential is only supported when running in Node.js.'
+    }
+
+    const
+      tenantId = process.env.AZURE_TENANT_ID,
+      clientId = process.env.AZURE_CLIENT_ID,
+      clientSecret = process.env.AZURE_CLIENT_SECRET;
```

Comment:
If these values changed when the user application is running, then the subsequent `getToken()` call will still be using the old credential

Reply:
Typically environment variables wouldn't change while the application is running unless someone intentionally set `process.env.*` fields after the `EnvironmentCredential` gets created.  Not sure what the correct expectation should be about whether changes should be picked up each time `getToken` is called.  @schaabs?

Reply:
I think that capturing these once per process is the right thing to do to remove the overhead of constantly having to read the environment for something which in most cases will not change.   If users want more dynamic behavior there are other credential types which would allow them to accomplish this.